### PR TITLE
Add Create Release GH Action

### PR DESCRIPTION
**Pull request by sachinsharmaepisource**
### Problem: 
When the master branch and develop branch are merged and current version > last version, than a new release is created with description as embedded PRs Titles.

Requirements:
- Python (PyGithub)
- GH Action (YML)
- DockerFile

### Flowchart
Feature branch -(merge)> Develop branch -(merge)> Master branch
----------------------------------------------------------- Here GH Action Initiates

### Key Features:

- Extracts last_version from latest release and current_version from version.ini file
- Compare last_version and current_version
> If current_version > last_version:             Action will create a new release
> Else:                                                          No change

### CODE REVIEW

File: .github/actions/createRelease/main.py [hunk 1]
```diff
@@ -0,0 +1,147 @@
+import os
+import sys
+from github import Github
+from datetime import datetime
+
+'''
+  requiremnet:
```
Comment (Prakshi-epi) [inline .github/actions/createRelease/main.py:7]: spell check

File: .github/actions/createRelease/main.py [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+      if is_new_merge_with_main():
+        if is_last_release_draft():
+          update_last_draft_release()
+        else:
+          create_new_draft_release()
+    else:
+      the current_version is smaller to last version, INVALID no new release.
+'''
+
+def get_datetime_now():
```
Comment (Prakshi-epi) [inline .github/actions/createRelease/main.py:28]: @sachinsharmaepisource Docstring is missing in all the functions. Refer to this guide on how to add a docstring to your code: https://numpydoc.readthedocs.io/en/latest/format.html

Also, add inline comments wherever necessary. Inline comments are used to describe your custom logic or some additional case you have handled in your code.

File: .github/actions/createRelease/main.py (hunk 1 above)
Comment (sachinsharmaepisource) [reply .github/actions/createRelease/main.py]: corrected 👍

File: .github/actions/createRelease/main.py (hunk 2 above)
Comment (sachinsharmaepisource) [reply .github/actions/createRelease/main.py:29]: Added the docStrings and inline comments 🚀

File: .github/actions/createRelease/main.py [hunk 3]
```diff
@@ -0,0 +1,327 @@
...
+      elif current_version == last_version:
+        if is_new_merge_with_main():
+          if is_last_release_draft():
+            update_last_draft_release()
+          else:
+            create_new_draft_release()
+      else:
+        the current_version is smaller to last version, INVALID no new release.
+'''
+
```
Comment (Prakshi-epi) [inline .github/actions/createRelease/main.py:28]: @sachinsharmaepisource Rewrite this script using Python Classes and Objects. The code looks clean but needs refactoring in terms of structure.

**Review by Prakshi-epi — CHANGES_REQUESTED** [review]
Please have a look at comments

File: .github/actions/createRelease/main.py (hunk 3 above)
Comment (sachinsharmaepisource) [reply .github/actions/createRelease/main.py:28]: Hi Prakshi, I had updated the code with class and object structure.

File: .github/actions/createRelease/main.py [hunk 4]
```diff
@@ -0,0 +1,309 @@
...
+    
+  def get_start_date_of_latest_release(self):
+    '''
+      Parameters
+      ----------
+          None
+      Logic
+      ----------
+      The start_date will store the date from
+        If the latest release exists:
```
Comment (Prakshi-epi) [inline .github/actions/createRelease/main.py:113]: Please cross-check this statement. I think you mean `If the latest release doesn't' exists`

File: .github/workflows/main.yml [hunk 5]
```diff
@@ -0,0 +1,26 @@
+name: Create Release
+
+on:
```
Comment (Prakshi-epi) [inline .github/workflows/main.yml:3]: Let's keep the condition to trigger the GitHub action on every commit in master branch
```
on:
  push:
    branches: [ master ]
```

Comment (sachinsharmaepisource) [reply .github/workflows/main.yml:5]: Updated with corrections mentioned

**Pull request closed without merging**